src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Marshaling/JSMarshalerArgument.Exception.cs

```diff
@@ -66,10 +66,16 @@ public unsafe void ToJS(Exception? value)
                 var jse = cpy as JSException;
                 if (jse != null && jse.jsException != null)
                 {
-                    ObjectDisposedException.ThrowIf(jse.jsException.IsDisposed, value);
-#if FEATURE_WASM_THREADS
-                    JSObject.AssertThreadAffinity(value);
-                    var ctx = jse.jsException.ProxyContext;
+                    var jsException = jse.jsException;
+#if !FEATURE_WASM_THREADS
+                    ObjectDisposedException.ThrowIf(jsException.IsDisposed, value);
+#else
+                    var ctx = jsException.ProxyContext;
+                    lock (ctx)
```

Comment:
This pattern might call for a `JSObject.ThrowIfDisposed` that does thread safety correctly in one spot